expyfun/visual/_visual.py

```diff
@@ -1111,27 +1094,31 @@ def set_pos(self, pos, units='norm', center=True):
         self._pos_unit = units
         self._pos_centered = center
 
+    def _draw(self):
+        self._texture = self._player.get_texture()
+        self._scale_texture()
+        self._texture.blit(*self._actual_pos)
+
     def draw(self):
         """Draw the video texture to the screen buffer."""
         self._player.update_texture()
         # detect end-of-stream to prevent pyglet from hanging:
         if not self._eos:
             if self._visible:
-                self._texture = self._player.get_texture()
-                self._scale_texture()
-                self._texture.blit(*self._actual_pos)
+                self._draw()
         else:
             self._finished = True
             self.pause()
         self._ec.check_force_quit()
 
     def set_visible(self, show, flip=False):
-        """Show/hide the video frame"""
+        """Show/hide the video frame."""
         if show:
             self._visible = True
+            self._draw()
         else:
-            self._texture.blit_into(self._background, 0, 0, 0)
             self._visible = False
+            self._ec.flip()
         if flip:
             self._ec.flip()
```

Comment:
this is super annoying, but for the life of me I can't get `set_visible(False)` to work without doing **two** flips (related to earlier comment about not being able to alter the `ImageData` in an existing pyglet `TextureRegion`).  Ideas for alternate approaches welcome.